### ISSUE TITLE
"C_Cpp.intelliSenseEngine": "Default" not working as expected

### ISSUE THREAD

**Issue by atavern1** [microsoft/vscode-cpptools#786]
I'm getting red squiggles indicating that a number of identifiers are undefined. I'm working from a Ubuntu machine and I've set up a c_cpp_properties.json as below (I removed the Mac and Windows config objects for brevity):

```
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "${workspaceRoot}",
                "/usr/include/c++/5.4.0",
                "/usr/local/include",
                "/usr/include",
                "/home/anthony/Programs/playground/code/core/includes"
            ],
            "defines": [],
            "browse": {
                "path": [
                    "/usr/include/c++/5.4.0",
                    "/usr/local/include",
                    "/usr/include",
                    "/home/anthony/Programs/playground/code/core/includes"
                ],
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": ""
            }
        }
    ]
}
```

I've also set up a user-setting override of "C_Cpp.intelliSenseEngine" to "Default". It seems that this is the way to enable experimental semantic autocomplete after doing some digging around the internet.

After setting up my project in this way, I'm getting red squiggles underneath uses of cout as well as non standard library includes like glfwInit letting me know these identifiers are not defined. The thing is, I'm sure that the configuration given above correctly points to the declaration files. 

What's weirder to me is that using 'Ctrl + click' on cout or glfwInit brings me immediately to their declarations. VS Code seems to have no problem when it comes to providing the click through functionality.

When I switch VS Code back to the Tag Parser method the red squiggles go away. 

I've seen some bugs related to unexpected squiggles (https://github.com/Microsoft/vscode-cpptools/issues/765) and (https://github.com/Microsoft/vscode-cpptools/issues/757) but I don't believe this is a related issue after reading through them.

Any guidance on what's wrong with the setup I've got is appreciated. I'd be glad to give more details on my setup if needed.

**Comment by sean-mcmanus** [comment]
Setting IntelliSenseEngine to Default does not yet enable semantic autocomplete (we're working on adding that next)...it just does errorSquiggles and hover now. The includePath setting is non-recursive, unlike the browse.path setting, which is. If you search through the #includes until you find the 1st missing file, that will tell you what file is really missing (we're in the process of adding this reporting automatically). You might be missing the folder `/usr/include/x86_64-linux-gnu/c++/5.4.0` and `/usr/include/x86_64-linux-gnu` (we have fix for that, which will add that to the default path of a new c_cpp_properties.json)...or you might need `/usr/lib/clang/<version>/include`. Are you missing system files or your project headers?

**Comment by dtasev** [comment]
Good to hear semantic autocomplete is in the works

**Comment by bobbrow** [comment]
@atavern1, were you able to get your c_cpp_properties.json file working?  If not, I recommend querying your compiler for the system include path that it uses and pasting that into the json file.
e.g. `clang -v -E -x c++ - < /dev/null`

**Comment by bobbrow** [comment]
@atavern1, if you run the clang command above and copy the include paths into your c_cpp_properties.json file, that should resolve the majority of your squiggles.  If that does not resolve your issue, please reopen and provide some additional details.

@DTasev, semantic member list suggestions (autocomplete) is available in 0.12.0.
